### PR TITLE
PlanningApplicationSearch

### PR DESCRIPTION
**Pull request by EGiataganas**
### Description of change

PlanningApplicationSearch

### Story Link

https://trello.com/c/ekD8BxM5

### CODE REVIEW

File: spec/services/planning_application_search_spec.rb [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+  let(:search) { described_class.new(params) }
+
+  let!(:local_authority) { create(:local_authority, :default) }
+
+  let!(:assessor) do
+    create(:user, :assessor, local_authority:)
+  end
+
+  before do
+    allow_any_instance_of(described_class).to receive(:current_user).and_return(assessor)
```
Comment (benjamineskola) [inline spec/services/planning_application_search_spec.rb:15]:
```suggestion
    allow(Current).to receive(:user).and_return(assessor)
```

I think it's probably bad practice to stub the class that's being tested. It looks like the `current_user` method just returns `Current.user`, so I think we should be able to stub that instead?

File: spec/services/planning_application_search_spec.rb [hunk 2]
```diff
@@ -0,0 +1,300 @@
...
+
+      context "when is search with exclude_others" do
+        let(:params) do
+          ActionController::Parameters.new(
+            { view: "mine" }
+          )
+        end
+
+        it "returns correct planning applications" do
+          expect(search.call).to eq([planning_application3])
```
[the remark this answers is not in the document]
Comment (benbaumann95) [reply spec/services/planning_application_search_spec.rb:79]: Yes I think we should be returning applications where the `current user` = `planning_application.user` and any other planning applications where `planning_application.user_id` is `nil`.

File: spec/services/planning_application_search_spec.rb (hunk 1 above)
Comment (benbaumann95) [reply spec/services/planning_application_search_spec.rb:15]: In some specs we just do `before { Current.user = assessor }`

File: spec/services/planning_application_search_spec.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (benbaumann95) [reply spec/services/planning_application_search_spec.rb:79]: It's associated with the user and null users (i.e. unassigned planning applications)

[the remark this answers is not in the document]
Comment (EGiataganas) [reply spec/services/planning_application_search_spec.rb]: I've fixed the spec and added an additional spec for that

**Review by cdccollins — APPROVED** [review]

**Review by benbaumann95 — APPROVED** [review]

**Pull request merged**